main/streams/filter.c

```diff
@@ -284,6 +284,12 @@ PHPAPI void php_stream_filter_free(php_stream_filter *filter)
 
 PHPAPI int php_stream_filter_prepend_ex(php_stream_filter_chain *chain, php_stream_filter *filter)
 {
+	if (chain->nb_filters >= PHP_STREAM_MAX_FILTERS) {
+	    php_error_docref(NULL, E_ERROR, "Unable to apply filter, maximum number (%d) reached", PHP_STREAM_MAX_FILTERS);
```

Comment:
We should avoid introducing new fatal errors if possible. We could use an exception here.